### PR TITLE
latest pkg:source_gen, and a lot of deprecation cleanup

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

**Comment by kevmoo** [comment]
@natebosch – yes, this is a bear. Happy to split it up if it'll help...

File: json_serializable/build.yaml [hunk 1]
```diff
@@ -2,8 +2,8 @@
 builders:
   json_serializable:
     target: ":json_serializable"
-    import: "package:json_serializable/builder.dart"
-    builder_factories: ["jsonSerializable"]
+    import: "package:json_serializable/json_serializable.dart"
+    builder_factories: ["createBuilderForJsonSerializable"]
```
Comment (natebosch) [inline json_serializable/build.yaml]: The changes in this file move away from the conventions we've been trying to establish. Perhaps I should document them somewhere...

Importing from `/builder.dart` is a signal that the code in here is meant more for the build system than for end users - it's good IMO to keep the builder separate from the annotations.

Having a noun name for the factory makes the build.yaml look a little nicer.

File: json_serializable/lib/src/builder.dart [hunk 2]
```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
```
Comment (natebosch) [inline json_serializable/lib/src/builder.dart]: 2018

File: json_serializable/lib/src/builder.dart [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import 'package:logging/logging.dart';
+
+import 'json_part_builder.dart';
+
+// TODO: until we can use `log` here - github.com/dart-lang/build/issues/1223
+final _logger = new Logger('json_serializable');
+
+/// Supports `package:build_runner` creation and configuration of `build_cli`.
+///
+/// Not meant to be invoked by hand-authored code.
```
Comment (natebosch) [inline json_serializable/lib/src/builder.dart]: which is why I like to have it in a separate library 😁

File: json_serializable/build.yaml (hunk 1 above)
Comment (kevmoo) [reply json_serializable/build.yaml]: Hrm...k

File: json_serializable/lib/src/builder.dart (hunk 3 above)
Comment (kevmoo) [reply json_serializable/lib/src/builder.dart]: Hrm...k

**Comment by kevmoo** [comment]
@natebosch PTAL

**Pull request merged**